### ISSUE TITLE
Получить полигоны парков из ЯНдекс Карт

### ISSUE THREAD

**Issue by Malaxov16** [NorthGateVologda/NorthGateBackend#35]

**Comment by Malaxov16** [comment]
Добавил координаты полигонов в форме прямоугольника в формате: 2 угла НЛ - нижний левый, ВП -верхний правый. 
Использовались API Яндекс карт https://yandex.ru/dev/geosearch/doc/ru/request.

Ответ возвращается в объекте. **Интересующий ключ: boundedBy**
Границы области, в которой предположительно находятся искомые объекты.

Задается в виде координат левого верхнего и правого нижнего углов области. Координаты указываются в последовательности «широта, долгота».

Границы области определяются сервисом автоматически.


**Парки с полигонами расположения:**

[Объекты парки.xlsx](https://github.com/NorthGateVologda/NorthGateBackend/files/12251307/default.xlsx)

**Comment by Malaxov16** [comment]
@olgapshen эти координаты нужно загрузить в БД к имеющимся паркам

**Comment by sshiae** [comment]
Крутяк!!!

**Comment by Marwin887** [comment]
@Malaxov16 @olgapshen Может координаты занести в один столбце указав их тип в другом столбце верх/низ, потом соотнести с полигонами и если верхняя и нижняя точка соотноситься с полигоном то брать только нижнюю точку (можно и верхнюю, это на вкус и цвет), т.о. часть парков получится в двух полигонах.

**Comment by olgapshen** [comment]
> Добавил координаты полигонов в форме прямоугольника в формате: 2 угла НЛ - нижний левый, ВП -верхний правый. Использовались API Яндекс карт https://yandex.ru/dev/geosearch/doc/ru/request.

@Malaxov16 ты бы не мог всё же залить свои труды в репозиторий. ЭТО ОЧЕНЬ, ОЧЕНЬ важно!!! По следующим причинам:

1. Возможно ты уйдёшь в отпуск, или какие то непредвиденные обстоятельства и нам нужно будет оперативно использовать твои наработанные инструменты
2. Имея готовый код на любом языке намного проще и быстрее внедрить его в наш бакенд

@sshiae @lvapl как вы думаете? Стоит ли заливать все наработки в репозиторий?

**Comment by olgapshen** [comment]
Я скоро залью данные в базу.

**Comment by sshiae** [comment]
> > Добавил координаты полигонов в форме прямоугольника в формате: 2 угла НЛ - нижний левый, ВП -верхний правый. Использовались API Яндекс карт https://yandex.ru/dev/geosearch/doc/ru/request.
> 
> @Malaxov16 ты бы не мог всё же залить свои труды в репозиторий. ЭТО ОЧЕНЬ, ОЧЕНЬ важно!!! По следующим причинам:
> 
> 1. Возможно ты уйдёшь в отпуск, или какие то непредвиденные обстоятельства и нам нужно будет оперативно использовать твои наработанные инструменты
> 2. Имея готовый код на любом языке намного проще и быстрее внедрить его в наш бакенд
> 
> @sshiae @lvapl как вы думаете? Стоит ли заливать все наработки в репозиторий?

Считаю, что стоит

**Comment by olgapshen** [comment]
> @olgapshen эти координаты нужно загрузить в БД к имеющимся паркам

А в какой таблице у нас уже имеются парки?

**Comment by olgapshen** [comment]
Ребята @Marwin887 @Malaxov16 @lvapl @sshiae, как создаёте таблицу, пожалуйста сразу выдавайте на неё права всей команде... я пока не разобралась с группами и ролями, но просто добавьте на закладке security остальных четверых членов команды.

![изображение](https://github.com/NorthGateVologda/NorthGateBackend/assets/13800244/0f26349e-2f8b-46b6-b5eb-20f03424e79a)

**Comment by olgapshen** [comment]
Я залью пока отдельно, потом если что обогатим.

**Comment by Malaxov16** [comment]
@olgapshen не смогтотрыть даталенс по ссылке, которую ты прикладывала в комментариях к другой задаче сегодня. Пишет, что у меня нет доступа. Что мне нужно сделать, чтобы получить доступ?

**Comment by olgapshen** [comment]
> @olgapshen не смогтотрыть даталенс по ссылке. Пишет, что у меня нет доступа. Что мне нужно сделать, чтобы получить доступ?

О! :) Наконец то :) Давно же я слала всем приглашения :) только ты не одобрил. Посмотри почту, которая указанна в файле ресурсов.

**Comment by olgapshen** [comment]
![изображение](https://github.com/NorthGateVologda/NorthGateBackend/assets/13800244/74c93ca0-0e4f-4446-8aec-cfa7bcb3b610)

![изображение](https://github.com/NorthGateVologda/NorthGateBackend/assets/13800244/e9676b1e-b439-40a7-9ef3-d6b865bd7712)

**Comment by olgapshen** [comment]
@Malaxov16 отправила повторно.

![изображение](https://github.com/NorthGateVologda/NorthGateBackend/assets/13800244/9ff54505-85bd-4357-91cb-26cc762b8e4f)

**Comment by olgapshen** [comment]
@Malaxov16 https://github.com/NorthGateVologda/NorthGateWiki/pull/18

**Comment by Malaxov16** [comment]
@olgapshen по ссылке в письме прошел, приглашение принял, но все равно не пускает ) Что еще не дожал?))

**Comment by olgapshen** [comment]
> @olgapshen по ссылке в письме прошел, приглашение принял, но все равно не пускает ) Что еще не дожал?))

Нужно было выдать ещё роли. Выдала :) Теперь заходит.

**Comment by Marwin887** [comment]
@Malaxov16 Андрюха посмотри у тебя в файле с полигонами вот эти координаты не правильные 

![Image](https://github.com/NorthGateVologda/NorthGateBackend/assets/136377502/2a8de252-0562-41ff-8b0e-48776741202a)

**Comment by Malaxov16** [comment]
@olgapshen @Marwin887 обновил координаты для двух парков, у которых некорректно определились полигоны. Но это, конечно, тот еще костыль)) я просто по карте руками координаты 2х углов взял, чтобы были.

Оля, в таблице parks_lens значения в столбцах max, min обновил. Пересчитай, пож та, столбец geo_lens. Я не очень понял какой процедурой он собирается.

**Comment by olgapshen** [comment]
> @olgapshen @Marwin887 обновил координаты для двух парков, у которых некорректно определились полигоны. Но это, конечно, тот еще костыль)) я просто по карте руками координаты 2х углов взял, чтобы были.

Нет, нет, это не костыль. Я сама кстати это хотела предложить. Это как бы "экспертный метод", не многим уступающий парсингу по сравнению с использованием API.

**Comment by olgapshen** [comment]
> @olgapshen @Marwin887 обновил координаты для двух парков, у которых некорректно определились полигоны. Но это, конечно, тот еще костыль)) я просто по карте руками координаты 2х углов взял, чтобы были.
> 
> Оля, в таблице parks_lens значения в столбцах max, min обновил. Пересчитай, пож та, столбец geo_lens. Я не очень понял какой процедурой он собирается.

Я сохранила этот код в репозитории NothGateDBScripts (пока в виде PR), но так же в файле enrich_parks в PGAdmin.
Тебе надо только его запустить, и всё пересчитается. Я запущу сейчас, но думаю ты сможешь его потом в дальнейшем использовать.

![изображение](https://github.com/NorthGateVologda/NorthGateBackend/assets/13800244/01f90325-0168-40c9-a066-0204f22dcd3d)

**Comment by olgapshen** [comment]
Запустила. Выглядит красивее :)

![изображение](https://github.com/NorthGateVologda/NorthGateBackend/assets/13800244/1ea3f712-3b1d-4ef6-8dc2-401548df4c19)
